common/persistence/sql/sqlplugin/sqlite/driver.go

```diff
@@ -28,21 +28,34 @@ package sqlite
 
 import (
 	"errors"
+	"regexp"
 
 	"modernc.org/sqlite"
 	sqlite3 "modernc.org/sqlite/lib"
 )
 
 const (
-	goSqlDriverName    = "sqlite"
-	sqlConstraintCodes = sqlite3.SQLITE_CONSTRAINT | sqlite3.SQLITE_CONSTRAINT_PRIMARYKEY | sqlite3.SQLITE_CONSTRAINT_UNIQUE
+	goSqlDriverName       = "sqlite"
+	sqlConstraintCodes    = sqlite3.SQLITE_CONSTRAINT | sqlite3.SQLITE_CONSTRAINT_PRIMARYKEY | sqlite3.SQLITE_CONSTRAINT_UNIQUE
+	sqlTableExistsPattern = "SQL logic error: table .* already exists \\(1\\)"
 )
 
-func (mdb *db) IsDupEntryError(err error) bool {
+var sqlTableExistsRegex = regexp.MustCompile(sqlTableExistsPattern)
+
+func (*db) IsDupEntryError(err error) bool {
 	var sqlErr *sqlite.Error
 	if errors.As(err, &sqlErr) {
 		return sqlErr.Code()&sqlConstraintCodes != 0
 	}
 
 	return false
 }
+
+func IsTableExistsError(err error) bool {
+	var sqlErr *sqlite.Error
+	if errors.As(err, &sqlErr) {
+		return sqlTableExistsRegex.MatchString(sqlErr.Error())
```

Comment:
Is it the only way? Parse error message?

Reply:
I don't see a problem here.